apps/devhub/tasks.py

```diff
@@ -272,17 +272,27 @@ def run_validator(path, for_appversions=None, test_all_tiers=False,
             path = temp.name
 
         with statsd.timer('devhub.validator'):
-            return validate(path,
-                            for_appversions=for_appversions,
-                            format='json',
-                            # When False, this flag says to stop testing after
-                            # one tier fails.
-                            determined=test_all_tiers,
-                            approved_applications=apps,
-                            spidermonkey=settings.SPIDERMONKEY,
-                            overrides=overrides,
-                            compat_test=compat,
-                            listed=listed)
+            result = validate(path,
+                              for_appversions=for_appversions,
+                              format='json',
+                              # When False, this flag says to stop testing
+                              # after one tier fails.
+                              determined=test_all_tiers,
+                              approved_applications=apps,
+                              spidermonkey=settings.SPIDERMONKEY,
+                              overrides=overrides,
+                              compat_test=compat,
+                              listed=listed)
+
+        result_kind = 'success' if result['errors'] == 0 else 'failure'
```

Comment:
What are you trying to count here? `errors` are the kind of errors that simply prevent submitting the file, I believe, like a missing install.rdf for example.

There's a few entries that may be useful here, see an example validation result from https://github.com/mozilla/olympia/blob/master/apps/constants/base.py#L378
- warnings
- success
- passed_auto_validation